runtime/src/accounts_db.rs

```diff
@@ -4396,11 +4396,14 @@ impl AccountsDb {
         if is_ancient(accounts) {
             // randomly shrink ancient slots
             // this exercises the ancient shrink code more often
-            if self.is_candidate_for_shrink(storage, true)
-                || (can_randomly_shrink
-                    && thread_rng().gen_range(0, 100) == 0
-                    && is_ancient(accounts))
-            {
+            let written_bytes = storage.written_bytes();
+            let is_candidate = if written_bytes > 0 {
+                let alive_ratio = (storage.alive_bytes() as u64) * 100 / written_bytes;
+                alive_ratio < 90
```

Comment:
Would making this `90` configurable make this more testable? I'm not sure it would, but think it's better to just have a named constant for this to make it very clear what this 90 represents.

Reply:
I doubt 90 is the final answer. There are several tradeoffs I'm worried about. I'm not sure what the final answer is yet. This is like shrink ratio, which is already configurable. But with shrink ratio, we do things a bit differently. And, the accounts in ancient append vecs by definition are rarely touched, which means we should treat things differently. Shrinking every time is definitely too inefficient. This will only have an effect when the cli arg is turned on.